compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -219,14 +219,6 @@ fn check_opaque(tcx: TyCtxt<'_>, id: hir::ItemId) {
         return;
     };
 
-    // HACK(jynelson): trying to infer the type of `impl trait` breaks documenting
-    // `async-std` (and `pub async fn` in general).
-    // Since rustdoc doesn't care about the concrete type behind `impl Trait`, just don't look at it!
-    // See https://github.com/rust-lang/rust/issues/75100
-    if tcx.sess.opts.actually_rustdoc {
-        return;
```

Comment:
Are there tests for this code path? If not, can you please double check that documenting async-std still works after this change? Make sure to enable all the features enabled on doc's.rs (you can use doc-like-docs.rs in @Nemo157 's dotfiles).

Reply:
Yes, there are tests. I'll run the try build on async-std once it finishes. Seems like `rustup toolchain link` on local builds doesn't work for rustdoc as `cargo +linked_thingy doc` can't find libcore and libstd

Reply:
That should work - try `x build rustdoc library`, or just `x build` without arguments.